diff --git a/systray_darwin.m b/systray_darwin.m
index 884fa43..7dd9500 100644
--- a/systray_darwin.m
+++ b/systray_darwin.m
@@ -74,6 +74,12 @@ withParentMenuId: (int)theParentMenuId
   systray_ready();
 }
 
+extern signed char lbs__ui__handle_app_reopen(signed char flag);
+
+- (BOOL)applicationShouldHandleReopen:(NSApplication *)sender hasVisibleWindows:(BOOL)flag {
+  return lbs__ui__handle_app_reopen(flag);
+}
+
 - (void)applicationWillTerminate:(NSNotification *)aNotification
 {
   systray_on_exit();
