diff --git a/src/syscall/exec_windows.go b/src/syscall/exec_windows.go
index 45295dedff..26ecf339ec 100644
--- a/src/syscall/exec_windows.go
+++ b/src/syscall/exec_windows.go
@@ -117,7 +117,7 @@ func makeCmdLine(args []string) string {
 // terminated strings followed by a nil.
 // Last bytes are two UCS-2 NULs, or four NUL bytes.
 // If any string contains a NUL, it returns (nil, EINVAL).
-func createEnvBlock(envv []string) (*uint16, error) {
+func CreateEnvBlock(envv []string) (*uint16, error) {
 	if len(envv) == 0 {
 		return &utf16.Encode([]rune("\x00\x00"))[0], nil
 	}
@@ -405,7 +405,7 @@ func StartProcess(argv0 string, argv []string, attr *ProcAttr) (pid int, handle
 		}
 	}
 
-	envBlock, err := createEnvBlock(attr.Env)
+	envBlock, err := CreateEnvBlock(attr.Env)
 	if err != nil {
 		return 0, 0, err
 	}
