diff --git a/packages/cardano-services/src/Http/HttpServer.ts b/packages/cardano-services/src/Http/HttpServer.ts
index 25da7a37ab..a8c04063d8 100644
--- a/packages/cardano-services/src/Http/HttpServer.ts
+++ b/packages/cardano-services/src/Http/HttpServer.ts
@@ -175,7 +175,9 @@ export class HttpServer extends RunnableModule {
     };
 
     let handler: keyof typeof handlers;
-    for (handler in handlers) this.app.use(path.join(versionPath, handler), handlers[handler]);
+    // Note: `.replaceAll(path.sep, '/')` is needed on Windows:
+    for (handler in handlers)
+      this.app.use(path.join(versionPath, handler).replaceAll(path.sep, '/'), handlers[handler]);
   }
 
   static sendJSON<ResponseBody>(
@@ -268,7 +270,10 @@ export class HttpServer extends RunnableModule {
         includePath: true,
         promRegistry,
         ...this.#config.metrics?.options,
-        metricsPath: path.join(versionPath, this.#config.metrics?.options?.metricsPath || '/metrics')
+        // Note: `.replaceAll(path.sep, '/')` is needed on Windows:
+        metricsPath: path
+          .join(versionPath, this.#config.metrics?.options?.metricsPath || '/metrics')
+          .replaceAll(path.sep, '/')
       })
     );
     this.logger.info(
diff --git a/packages/cardano-services/test/Http/HttpServer.test.ts b/packages/cardano-services/test/Http/HttpServer.test.ts
index 7e6b57879c..65f72ed0c3 100644
--- a/packages/cardano-services/test/Http/HttpServer.test.ts
+++ b/packages/cardano-services/test/Http/HttpServer.test.ts
@@ -188,13 +188,15 @@ describe('HttpServer', () => {
         {
           body: { bigint: 23n, check: 'ok', test: 42 },
           method: 'POST',
-          path: path.join(someServiceVersionPath, 'stake-pool', 'echo'),
+          // Note: `.replaceAll(path.sep, '/')` is needed on Windows:
+          path: path.join(someServiceVersionPath, 'stake-pool', 'echo').replaceAll(path.sep, '/'),
           query: {}
         },
         {
           body: {},
           method: 'GET',
-          path: path.join(someServiceVersionPath, 'stake-pool', 'echo'),
+          // Note: `.replaceAll(path.sep, '/')` is needed on Windows:
+          path: path.join(someServiceVersionPath, 'stake-pool', 'echo').replaceAll(path.sep, '/'),
           query: { bigint: '23', check: 'ok', test: '42' }
         }
       ]);
