diff --git a/src/syscall/exec_windows.go b/src/syscall/exec_windows.go
index 45295dedff..cae07b0433 100644
--- a/src/syscall/exec_windows.go
+++ b/src/syscall/exec_windows.go
@@ -250,6 +250,8 @@ type SysProcAttr struct {
 	NoInheritHandles           bool                // if set, each inheritable handle in the calling process is not inherited by the new process
 	AdditionalInheritedHandles []Handle            // a list of additional handles, already marked as inheritable, that will be inherited by the new process
 	ParentProcess              Handle              // if non-zero, the new process regards the process given by this handle as its parent process, and AdditionalInheritedHandles, if set, should exist in this parent process
+	StartupInfoCbReserved2     uint16
+	StartupInfoLpReserved2     *byte
 }
 
 var zeroProcAttr ProcAttr
@@ -410,6 +412,9 @@ func StartProcess(argv0 string, argv []string, attr *ProcAttr) (pid int, handle
 		return 0, 0, err
 	}
 
+	si.CbReserved2 = sys.StartupInfoCbReserved2
+	si.LpReserved2 = sys.StartupInfoLpReserved2
+
 	pi := new(ProcessInformation)
 	flags := sys.CreationFlags | CREATE_UNICODE_ENVIRONMENT | _EXTENDED_STARTUPINFO_PRESENT
 	if sys.Token != 0 {
diff --git a/src/syscall/types_windows.go b/src/syscall/types_windows.go
index 384b5b4f2c..497ccf7357 100644
--- a/src/syscall/types_windows.go
+++ b/src/syscall/types_windows.go
@@ -483,8 +483,8 @@ type StartupInfo struct {
 	FillAttribute uint32
 	Flags         uint32
 	ShowWindow    uint16
-	_             uint16
-	_             *byte
+	CbReserved2   uint16
+	LpReserved2   *byte
 	StdInput      Handle
 	StdOutput     Handle
 	StdErr        Handle
