diff --git a/server/ogmios.cabal b/server/ogmios.cabal
index fc0d752f1..44e3c7b8e 100644
--- a/server/ogmios.cabal
+++ b/server/ogmios.cabal
@@ -33,6 +33,10 @@ flag production
   manual: True
   default: False
 
+common maybe-unix
+  if !os(windows)
+    build-depends:      unix
+
 library
   exposed-modules:
       Ogmios
@@ -121,6 +125,8 @@ library
   ghc-options: -Wall -Wcompat -Widentities -Wincomplete-record-updates -Wincomplete-uni-patterns -Wpartial-fields -Wredundant-constraints -Wunused-packages
   build-tools:
       git
+  import:
+      maybe-unix
   build-depends:
       aeson
     , array
@@ -192,7 +198,6 @@ library
     , time-manager
     , transformers
     , typed-protocols
-    , unix
     , unordered-containers
     , vector
     , wai
diff --git a/server/src/Ogmios.hs b/server/src/Ogmios.hs
index c48cbadc8..75f51720b 100644
--- a/server/src/Ogmios.hs
+++ b/server/src/Ogmios.hs
@@ -2,6 +2,7 @@
 --  License, v. 2.0. If a copy of the MPL was not distributed with this
 --  file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE DerivingVia #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE TypeApplications #-}
@@ -104,6 +105,7 @@ import Ogmios.Options
 import Ogmios.Version
     ( version
     )
+#ifdef UNIX
 import System.Posix.Signals
     ( Handler (..)
     , installHandler
@@ -111,6 +113,7 @@ import System.Posix.Signals
     , raiseSignal
     , softwareTermination
     )
+#endif
 
 import qualified Control.Concurrent.Async as Async
 import qualified Control.Concurrent.STM.TBQueue as STM
@@ -167,9 +170,13 @@ application tracers = hijackSigTerm >> withDebouncer _10s (\debouncer -> do
 -- This function install handlers for SIGTERM and turn them into SIGINT.
 hijackSigTerm :: App ()
 hijackSigTerm =
+#ifdef UNIX
     liftIO $ void (installHandler softwareTermination handler empty)
   where
     handler = CatchOnce (raiseSignal keyboardSignal)
+#else
+    pure ()
+#endif
 
 --
 -- Environment
diff --git a/server/src/Ogmios/Options.hs b/server/src/Ogmios/Options.hs
index 52905f2f8..a4138904f 100644
--- a/server/src/Ogmios/Options.hs
+++ b/server/src/Ogmios/Options.hs
@@ -87,7 +87,7 @@ import Options.Applicative.Help.Pretty
 import Safe
     ( readMay
     )
-import System.FilePath.Posix
+import System.FilePath
     ( replaceFileName
     )
 
