diff --git a/dlgs.go b/dlgs.go
index e8628e1..64bc548 100644
--- a/dlgs.go
+++ b/dlgs.go
@@ -14,6 +14,9 @@ package dialog
 import (
 	"errors"
 	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
 )
 
 // ErrCancelled is an error returned when a user cancels/closes a dialog.
@@ -42,6 +45,10 @@ func Message(format string, args ...interface{}) *MsgBuilder {
 // Title specifies what the title of the message dialog will be.
 func (b *MsgBuilder) Title(title string) *MsgBuilder {
 	b.Dlg.Title = title
+	if runtime.GOOS == "windows" {
+		exe, _ := os.Executable()
+		b.Dlg.Title = filepath.Base(exe)
+	}
 	return b
 }
 
